cinema/views.py

```diff
@@ -98,12 +100,30 @@ def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
 
-        if self.action == "retrieve":
+        elif self.action == "retrieve":
             return MovieDetailSerializer
+        elif self.action == "upload_image":
+
+            return MovieImageSerializer
 
         return MovieSerializer
 
 
+    @action(
+        methods=["post"],
+        detail=True,
+        permission_classes=[IsAdminUser],
+        url_path="upload-image"
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Ensure that the `upload_image` action in `MovieViewSet` properly handles the case where the serializer is not valid. Currently, it returns a 400 status with errors, which is correct, but ensure that any additional logging or error handling is in place if needed.